storage/indexer/src/db_v2.rs

```diff
@@ -0,0 +1,407 @@
+                } else {
+                    for (_identifier, field) in &struct_value.value {
+                        self.parse_move_value(field)?;
+                    }
+                }
+            },
+
+            // there won't be tables in primitives
+            AnnotatedMoveValue::U8(_) => {},
+            AnnotatedMoveValue::U16(_) => {},
+            AnnotatedMoveValue::U32(_) => {},
+            AnnotatedMoveValue::U64(_) => {},
+            AnnotatedMoveValue::U128(_) => {},
+            AnnotatedMoveValue::U256(_) => {},
+            AnnotatedMoveValue::Bool(_) => {},
+            AnnotatedMoveValue::Address(_) => {},
+            AnnotatedMoveValue::Bytes(_) => {},
+        }
+        Ok(())
+    }
+
+    fn save_table_info(&mut self, handle: TableHandle, info: TableInfo) -> Result<()> {
+        if self.get_table_info(handle)?.is_none() {
+            self.result.insert(handle, info);
+            if let Some(pending_items) = self.pending_on.remove(&handle) {
+                for bytes in pending_items.1 {
+                    self.parse_table_item(handle, &bytes)?;
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn is_table(struct_tag: &StructTag) -> bool {
+        struct_tag.address == AccountAddress::ONE
+            && struct_tag.module.as_ident_str() == ident_str!("table")
+            && struct_tag.name.as_ident_str() == ident_str!("Table")
+    }
+
+    /// Retrieves table information either from the in-memory results or from the database.
+    ///
+    /// This method first checks if the table information for the given handle exists in the
+    /// in-memory `result` Dashmap. If it is found, it returns the information directly from
+    /// there. If not, it fetches the table information from the database using the `IndexerAsyncV2`
+    /// instance. This approach of checking in-memory cache first improves performance by avoiding
+    /// unnecessary database reads.
+    ///
+    /// # Arguments
+    /// * `handle` - The table handle for which information is needed.
+    ///
+    /// # Returns
+    /// Returns `Result<Option<TableInfo>>`, which is the table information if available, or `None` if not.
+    fn get_table_info(&self, handle: TableHandle) -> Result<Option<TableInfo>> {
+        match self.result.get(&handle) {
+            Some(table_info) => Ok(Some(table_info.clone())),
+            None => self.indexer_async_v2.get_table_info(handle),
+        }
+    }
+
+    /// Finishes the parsing process and writes the parsed table information to a SchemaBatch.
+    ///
+    /// # Arguments
+    ///
+    /// * `batch` - A mutable reference to the SchemaBatch.
+    ///
+    /// # Returns
+    ///
+    /// Returns `Ok(true)` if the parsed table information is written to the SchemaBatch,
+    /// `Ok(false)` if there is no table information to write,
+    /// or an error if an error occurs during the writing process.
+    fn finish(self, batch: &mut SchemaBatch) -> Result<bool> {
```

Comment:
This is a poorly written function. As I see you're not even using the bool. This should just be Ok(()) then. 

But more importantly, why are we doing the pending_on cleanup in each batch? I'd say that it's cleaner to do this after the threads.join.